implement Vector operators __add__, __sub__, __mul__

justbuchanan opened the pull request.

adam-urbanczyk commented:
Looks good. One question though: do you have a specific use case for the Vector object in mind?
 
Also it would be nice to add some tests.

justbuchanan commented:
cqparts uses `Vector`s fairly extensively for moving between coordinate systems, among other things. Since several of the cadquery APIs accept and return `Vector` objects, it works out well to use them throughout. It would certainly be possible to use methods like `multiply` and `add` directly, but the operators are really convenient.

I added some test cases and implemented `__truediv__` to complete the basic four operators.

adam-urbanczyk commented:
Thanks for the contribution @justbuchanan
